### DISCUSSION TITLE
I cannot setup the Make connection dynamically

### DISCUSSION THREAD

**Discussion by GlennAtCallinetic** [Budibase/budibase#13874] [Help] [answered]
I am developing an automation when new rows are added to my table. Therefore, I want to setup an integration with Make. However, during the configuration I'm unable to bind my data dynamically to the payload. Also, the payload is sent as string instead as a JSON.
What am I missing here?

<img width="1094" alt="Screenshot 2024-06-06 at 20 32 00" src="https://github.com/Budibase/budibase/assets/103042444/1a2bea41-206f-4d1e-9e56-df5503e9582f">

**Comment by mikesealey** [comment]
Hey @GlennAtCallinetic, what's the output of your automation? There may be some clues hiding there...

**Reply by GlennAtCallinetic** [reply]
No output yet, as I’m still in the process of “determining the data structure” in Make. In practice, this process returns an HTTP 200 with the body “accepted.” But that’s not the final output, of course.

**Reply by mikesealey** [reply]
In the screenshot you posted on the right underneath "Step 1 - Make Integration" there are two tabs; input and output. Take a look in the Output tab. This might give you some idea of what's going wrong with the automation.

**Reply by GlennAtCallinetic** [reply]
This is the output. But it already starts with the payload I cannot build dynamically in Budibase.
```
{
"httpStatus": 400,
"response": "Invalid payload JSON",
"success": false
}
```

**Reply by mikesealey** [reply, marked as answer]
Hey @GlennAtCallinetic we've been looking into this together and come up with a solution.

We added a JS Scripting Block just before the Make integration, and used that to build our JSON object. 

Inside the JS block we built an object containing the keys and values required...
![image](https://github.com/Budibase/budibase/assets/39269380/1b371b5c-81ad-4488-9f93-a2da4aaad7e8)

...Then in the Make integration, we just pulled the value from the JS block...
![image](https://github.com/Budibase/budibase/assets/39269380/7528cd3f-d98a-43c1-9b67-0ce3d3e49fd3)

...Which resulted in a successful call, which we could see in our Make setup.

There is currently some work going on around the handlebar bindings in Automations, which may be the source of the issue. For the time being, please use this extra step to get you back on track.
